scico/admm.py

```diff
@@ -566,58 +565,49 @@ def u_init(self, x0: Union[JaxArray, BlockArray]):
         u_list = [snp.zeros(Ci.output_shape, dtype=Ci.output_dtype) for Ci in self.C_list]
         return u_list
 
-    def x_step(self, x):
-        r"""Update :math:`\mb{x}` by solving the optimization problem.
-
-        .. math::
-            \mb{x}^{(k+1)} = \argmin_{\mb{x}} \; f(\mb{x}) + \sum_i \frac{\rho_i}{2}
-            \norm{\mb{z}^{(k)}_i - \mb{u}^{(k)}_i - C_i \mb{x}}_2^2
+    def step(self):
+        r"""Perform a single ADMM iteration.
 
-        """
-        return self.subproblem_solver.solve(x)
+        The primary variable :math:`\mb{x}` is updated by solving the the
+        optimization problem
 
-    def z_and_u_step(self, u_list, z_list):
```

Comment:
I suspect the code was written this way so that the `z_and_u_step` could be jitted (jitted functions should be pure, no side effects). However, `z_and_u_step` was not actually jitted as far as I can tell. So I don't mind the refactor.

Reply:
That was my suspicion too. If we decide there is some benefit to jitting, we can refactor again, but for now this simpler structure makes more sense.